src/platform-implementation-js/dom-driver/gmail/views/gmail-thread-view.ts

```diff
@@ -675,6 +675,54 @@ class GmailThreadView {
     return new BasicButtonViewController(buttonOptions);
   }
 
+  addFooterButton(button: any) {
+    const footerParent = this._element.querySelector('.gA.gt.acV');
+    if (!footerParent) {
+      throw new Error('Subject wrapper element not found');
+    }
+
+    const buttonOptions = {
+      ...button,
```

Comment:
Is a shallow clone enough here? We mutate `buttonOptions.buttonView` below. Potentially we want a deep clone instead.